Author: Andreas Beckmann <anbe@debian.org>
Description: Select the correct nv-kernel.o blob for the target architecture
 This patch makes the build system select the correct one depending on the
 kernel architecture the module is built for.

--- a/nvidia/nvidia.Kbuild
+++ b/nvidia/nvidia.Kbuild
@@ -37,7 +37,9 @@
 # and needs to be re-executed.
 #
 
-NVIDIA_BINARY_OBJECT := $(src)/nvidia/nv-kernel.o_binary
+NVIDIA_BINARY_OBJECT-$(CONFIG_X86_64)	+= nv-kernel-amd64.o_binary
+NVIDIA_BINARY_OBJECT-$(CONFIG_PPC64)	+= nv-kernel-ppc64el.o_binary
+NVIDIA_BINARY_OBJECT := $(src)/nvidia/$(NVIDIA_BINARY_OBJECT-y)
 NVIDIA_BINARY_OBJECT_O := nvidia/nv-kernel.o
 
 quiet_cmd_symlink = SYMLINK $@
